.github/workflows/compile_ubuntu.yml

```diff
@@ -14,11 +14,11 @@ on:
 jobs:
   build_and_test:
     name: Build and Test Ubuntu
-    # strategy:
-    #   fail-fast: false
-    #   matrix:
-    #     version: [ubuntu22, ubuntu24]
-    runs-on: [runs-on,runner=8cpu-linux-x64,"image=ubuntu22-full-x64","run-id=${{ github.run_id }}"]
+    strategy:
+      fail-fast: false
+      matrix:
+        version: [ubuntu22, ubuntu24]
+    runs-on: [runs-on,runner=8cpu-linux-x64,"image=${{ matrix.version }}-full-x64","run-id=${{ github.run_id }}"]
```

Comment:
Can these be any docker image?

I'm thinking we should use the stock minimal ubuntu images so we catch all dependencies and have easy access to the full matrix of versions in between LTS and beyond. https://hub.docker.com/_/ubuntu

Reply:
Yes, we can test the available Ubuntu images on the docker. The runs-on can only find publicly available GitHub images, this has two drawbacks:
1. We depend on GitHub for adding newer Ubuntu LTS images
2. We can only use their images since custom image support is locked behind the GitHub self-hosted runners, which we have, but someone who forks might not.